js/date-filter.js

```diff
@@ -25,8 +25,13 @@ function DateFilter(elm) {
     oncomplete: this.validate.bind(this)
   });
 
+  if (this.$minDate.attr('data-min-date')) {
```

Comment:
It does? The nav link includes a `max_date` (which is there to hide expenditures that were incorrectly marked as happening in the future), but other than that I don't think we should be limiting by date automatically. (In principal I agree with you, though).